File: src/main/resources/static/entity/restaurant.js
export class Restaurant{
    userName;
    newUserName;
    passWord;
    newPassWord
    name;
    phone;
    city;
    address;
    managerName;
    managerSurName;
    balance;
    accountNo;
    longitude;
    latitude;
    hrFrom;
    hrTo;
    //lists
    foodMenu;
    attachmentMenu;
    subscription;
    orders;
    supportRequest;

    constructor(userName, newUserName, passWord, newPassWord, name, phone, address, city, managerName, managerSurName,
                balance, foodMenu, attachmentMenu, subscription, orders, supportRequest, accountNo, hrFrom, hrTo) {
        this.userName = userName;
        this.newUserName = newUserName;
        this.passWord = passWord;
        this.newPassWord = newPassWord;
        this.name = name;
        this.phone = phone;
        this.city = city;
        this.address = address;
        this.managerName = managerName;
        this.managerSurName = managerSurName;
        this.balance = balance;
        this.foodMenu = foodMenu;
        this.attachmentMenu = attachmentMenu;
        this.subscription = subscription;
        this.orders = orders;
        this.supportRequest = supportRequest;
        this.accountNo = accountNo;
        this.hrFrom = hrFrom;
        this.hrTo = hrTo;
    }
}

File: src/main/java/com/restaurant/restaurantordersystem/entity/Restaurant.java
package com.restaurant.restaurantordersystem.entity;

import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.JsonIdentityReference;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;
import jakarta.persistence.*;
import lombok.Data;

import java.io.Serializable;
import java.util.List;

@Data
@Entity
@Table
@JsonIdentityReference
@JsonIdentityInfo(generator = ObjectIdGenerators.PropertyGenerator.class , property = "id")
public class Restaurant implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String userName;
    transient public String newUserName;
    private String passWord;
    transient public String newPassWord;
    private String name;
    private String phone;
    private String address;
    private String city;
    private Double longitude;
    private Double latitude;
    private String managerName;
    private String managerSurName;
    private String balance;
    private Integer hrFrom;
    private Integer hrTo;
    private String AccountNo;
    private boolean suspended;

    @OneToMany(cascade = CascadeType.ALL)
    private List<FoodMenu> foodMenu;

    @OneToMany(cascade = CascadeType.ALL)
    private List<AttachmentMenu> attachmentMenu;

    @OneToMany(cascade = CascadeType.ALL)
    private List<FoodOrder> orders;

    @OneToMany(cascade = CascadeType.ALL)
    private List<Support> supportRequest;

    @OneToMany(cascade = CascadeType.ALL)
    private List<Subscription> subscriptions;

    transient private Auth admin;

    public void appendBalance(String amount){
        setBalance(String.valueOf(Long.parseLong(getBalance())+Long.parseLong(amount)));
    }
    public void withdrawBalance(String amount){
        setBalance(String.valueOf(Long.parseLong(getBalance())-Long.parseLong(amount)));
    }

}
